### DISCUSSION TITLE
Question about how to should I link my masters K8s in a cluster

### DISCUSSION THREAD

**Discussion by wardensfx** [gravitational/teleport#13965] [Q&A] [answered]
Hi,

I'm setting up a K8s HA baremetal cluster with three master and two nodes,
and I was wondering if I should make them communicate them through teleport ?

I first thought about machine-id, but is it capable to secure https communications instead of ssh ?

Or should I let direct connections , like between my masters on port 6443 ?

Thanks

**Comment by webvictim** [comment, marked as answer]
Machine ID doesn't currently provide application access, so I'd go with the traditional approach.

It's on the roadmap for a future version.
